indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamPartitions.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.seekablestream;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+
+import javax.validation.constraints.NotNull;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * class that encapsulates a map of partitionId -> sequenceNumber.
+ * To be backward compatible with both Kafka and Kinesis datasource metadata when
+ * deserializing json. Redundant constructor fields stream, topic and
+ * partitionSequenceNumberMap and partitionOffsetMap are created. Only one of topic, stream
+ * should have a non-null value and only one of partitionOffsetMap and partitionSequenceNumberMap
+ * should have a non-null value.
+ * <p>
+ * Redundant getters
+ * are used for proper Jackson serialization/deserialization when processing terminologies
+ * used by Kafka and kinesis (i.e. topic vs. stream)
+ *
+ * @param <PartitionType> partition id type
+ * @param <SequenceType>  sequence number type
+ */
+public class SeekableStreamPartitions<PartitionType, SequenceType>
+{
+  public static final String NO_END_SEQUENCE_NUMBER = "NO_END_SEQUENCE_NUMBER";
+
+  // stream/topic
+  private final String stream;
+  // partitionId -> sequence number
+  private final Map<PartitionType, SequenceType> map;
```

Comment:
IIRC, the instances of this class are logged by the overlord. Please use a more intuitive name like `partitionIdToSequenceNumberMap`.